lib/adapters/call-hierarchy-adapter.ts

```diff
@@ -0,0 +1,145 @@
+import type * as atomIde from "atom-ide-base"
+import Convert from "../convert"
+import * as Utils from "../utils"
+import { SymbolTag } from "../languageclient"
+import type { LanguageClientConnection, ServerCapabilities, CallHierarchyItem } from "../languageclient"
+import type { CancellationTokenSource } from "vscode-jsonrpc"
+import type { Point, TextEditor } from "atom"
+
+import OutlineViewAdapter from "./outline-view-adapter"
+
+/** Public: Adapts the documentSymbolProvider of the language server to the Outline View supplied by Atom IDE UI. */
+export default class CallHierarchyAdapter {
+  private _cancellationTokens: WeakMap<LanguageClientConnection, CancellationTokenSource> = new WeakMap()
+
+  /**
+   * Public: Determine whether this adapter can be used to adapt a language server based on the serverCapabilities
+   * matrix containing a callHierarchyProvider.
+   *
+   * @param serverCapabilities The {ServerCapabilities} of the language server to consider.
+   * @returns A {Boolean} indicating adapter can adapt the server based on the given serverCapabilities.
+   */
+  public static canAdapt(serverCapabilities: ServerCapabilities): boolean {
+    return !!serverCapabilities.callHierarchyProvider
```

Comment:
```suggestion
    return serverCapabilities.callHierarchyProvider !== undefined
```

Reply:
The type of `serverCapabilities.callHierarchyProvider` is `boolean | CallHierarchyOptions | CallHierarchyRegistrationOptions | undefined`, so I changed it to use `Boolean()`.